beps/bep_0007.rst

```diff
@@ -23,32 +24,31 @@ be interpreted as described in IETF `RFC 2119`_.
 
 .. _`RFC 2119`: http://tools.ietf.org/html/rfc2119
 
-Announce Parameter
-==================
+Announcing
+==========
+
+For each local IP address:
 
-The client MAY add an ``&ipv6=`` parameter to the HTTP GET request it sends
-to the tracker. The value is either an IPv6 endpoint (address and port) or
-just an IPv6 address. In the case where only an address is supplied, the IPv6
-port is assumed to be the same as specified by the ``&port=`` parameter.
+	1. the client intends to receive incoming peer connections over (listens on)
+	2. the client intends to publish to the tracker
+	3. that can route packets to the tracker
 
-The tracker SHOULD perform a NAT check on the IPv6 endpoint.
+an announce to the tracker SHOULD be made. Each announce should use the
+corresponding local IP address as the source address of the connection.
 
-In case the client contacts the tracker on an IPv6 interface, it may add
-an ``&ipv4=`` parameter with its IPv4 address or endpoint. The value MUST
-be either an IPv4 endpoint (address and port) or just an IPv4 address. If only
-an address is supplied, the port is assumed to be the same as the ``&port=``
-parameter.
+The source address is set on a socket using the ``bind()`` call, before connecting.
 
-The endpoints are encoded as strings as defined by `RFC 2732`_.
+This applies to TCP connections, for HTTP and HTTPS trackers, as well as UDP
+announces, to UDP trackers.
 
-.. _`RFC 2732`: http://tools.ietf.org/html/rfc2732
+If the tracker does not resolve to a an address family compatible with a source
+address, the client SHOULD NOT announce that address to the tracker.
```

Comment:
Well, it cannot, so the SHOULD NOT is moot, isn't it?

Reply:
yes, this is a pointless paragraph